src/main/java/io/github/syst3ms/skriptparser/sections/SecAsync.java

```diff
@@ -0,0 +1,62 @@
+package io.github.syst3ms.skriptparser.sections;
+
+import io.github.syst3ms.skriptparser.Main;
+import io.github.syst3ms.skriptparser.file.FileSection;
+import io.github.syst3ms.skriptparser.lang.CodeSection;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.Statement;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.log.SkriptLogger;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.parsing.ParserState;
+import io.github.syst3ms.skriptparser.util.ThreadUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Executes the code in the section asynchronously, meaning in another thread.
+ * Note that the next code that isn't part of the section (intended the same amount of times) will be executed in the current thread again.
+ * Only the code that is inside the section will be executed from another thread.
+ * This may cause some delay. If you don't know what this is, you probably don't need it.
+ *
+ * @name Async
+ * @type SECTION
+ * @pattern async[hronous[ly]]
+ * @since ALPHA
+ * @author Mwexim
+ */
+public class SecAsync extends CodeSection {
```

Comment:
This section is implemented nicely based on the current state of the API. In the future though, a better API could carry this sort of workload more efficiently.

Reply:
I'll wait until your suggestions tomorrow and we'll work it out from there (or you can merge it and change it later).

Reply:
Oh don't worry, this class could be updated rather smoothly with the system I have in mind.